### ISSUE TITLE
[iOS] FilePicker: File.Exist() not working and inconsistent behavior (read/write) | missing documentation

### ISSUE THREAD

**Issue by jp-weber** [dotnet/maui#8378]
### Description

I am developing an application that will access documents in a cloud storage through the FilePicker (Nextcloud with the file provider). As per the official instructions, I have added iCloud as a capability for the application/account provising. 

As I found out, the access to a file selected with the FilePicker (/privat/var/...) requires the call "StartAccessingSecurityScopedResource", which is used in PlatformOpenReadAsync().
https://github.com/dotnet/maui/blob/f8936a5c62d35eecd95d4345dcb30e1db2419d2d/src/Essentials/src/FileSystem/FileSystem.ios.cs#L89

After using the OpenReadAsync() function, access is possible with File.Read/Write. However, File.Exists() does not work. 

Especially if the file path is now saved and used when the application is started again, access to the file is no longer possible. Even if a FileResult with the file path is formed manually and the OpenReadAsync() function is called up.

```csharp
FileResult fileResult = new FileResult(dbDatafile.Path);
string datafileStr;
using (StreamReader reader = new StreamReader(await fileResult.OpenReadAsync()))
{
    datafileStr = reader.ReadToEnd();
}
//exception missing access rights
```
I only found this entry in the documentation. It would be nice if a best practice in the maui documentation explained a later access to the files.
https://github.com/MicrosoftDocs/xamarin-docs/blob/live/docs/ios/platform/document-picker.md

Also relevant issue #1944

### Steps to Reproduce

1. Select a file via FilePicker()
2. Call OpenReadAsync() on the result
3. Call File.Exist(result.FullPath) => false
4. Restart the app and create formed FileResult manually with the file path and call the OpenReadAsync() 
```csharp
FileResult fileResult = new FileResult(dbDatafile.Path);
string datafileStr;
using (StreamReader reader = new StreamReader(await fileResult.OpenReadAsync()))
{
    datafileStr = reader.ReadToEnd();
}
```
5. exception missing access rights

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15.5

### Did you find any workaround?

I will implement a custom method for iOS that calls the Start/StopAccessingSecurityScopedResource function for the file path. The access for File.Read/Write should work. But File.Exist() is still an open problem.

**Comment by rachelkang** [comment]
Hi, @jp-weber - thanks for opening this issue. You may find the documentation at https://docs.microsoft.com/xamarin/essentials/file-picker?tabs=ios to also be helpful.

As for the issue reported - are you experiencing this only on iOS? Is it working as expected on the other platforms?

**Comment by jp-weber** [comment]
Yes, I have only been able to detect the problem on iOS at the moment. Previously, everything worked perfectly under Android. What surprises me is that File.Exist() does not work with the path of the file from the FilePicker.

For the MAUI FilePicker I also used the doc: https://docs.microsoft.com/en-us/dotnet/maui/platform-integration/storage/file-picker?tabs=ios

**Comment by hevey** [comment]
+1 this issue. I have been able to store the path and re-create a file result and open on Mac Catalyst but when trying on iOS devices it fails (not accessible). 

It would be good to know whether manually creating a FileResult with a stored path to the file is expected to work, if not is there another way we should be getting access (FileSystemUtils which is used by the FilePicker isn't public) or do we need to roll our own solution

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with VSM 17.6.7 (build 417). Can repro this issue.
